kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/BaseOperation.java

```diff
@@ -922,13 +922,13 @@ public OperationInfo forOperationType(String type) {
   }
 
   @Override
-  public Deletable<Boolean> withGracePeriod(long gracePeriodSeconds)
+  public FilterWatchListDeletable<T, L, Boolean, Watch, Watcher<T>> withGracePeriod(long gracePeriodSeconds)
```

Comment:
Since, it works without breaking anything I am fine with it.

I don't see however, how it solves the two methods being mutually exclusive.

Reply:
`withGracePeriod` now return a [Deletable](https://static.javadoc.io/io.fabric8/kubernetes-client/4.6.1/io/fabric8/kubernetes/client/dsl/Deletable.html) and on it you can only call `delete`, you need to do an unsafe cast to be able to chain further methods (i.e. such as `withPropagationPolicy`)

Reply:
added a "demonstration" test for it